### PR TITLE
chore(refactor): centralize all providers `set` method

### PR DESCRIPTION
**Pull request by wa0x6e**
## 🧿 Current issues / What's wrong ?

We currently import all providers, then call each one of them with the same params. 
This leads to lot of duplicate code.

## 💊 Fixes / Solution

Centralize the `set` call, to avoid duplication.

This will enable in a later commit to remove the timing logic inside each providers, and add metrics.

## 🚧 Changes

- Add a provider factory, which will call the `set` function of a given list of providers, and return their promises.

## 🛠️ Tests

- Run `yarn test`. E2E test suite should cover enough ground.
- Alternatively, you can use postman to upload a file and json

### CODE REVIEW

File: src/upload.ts [hunk 1]
```diff
@@ -3,12 +3,9 @@ import express from 'express';
...
 import multer from 'multer';
 import sharp from 'sharp';
-import { rpcError, rpcSuccess } from './utils';
-import { set as setFleek } from './providers/fleek';
-import { set as setInfura } from './providers/infura';
-import { set as setPinata } from './providers/pinata';
-import { set as set4everland } from './providers/4everland';
 import { capture } from '@snapshot-labs/snapshot-sentry';
+import { rpcError, rpcSuccess } from './utils';
+import { IMAGE_PROVIDERS, default as set } from './providers/';
```
Comment (Todmy) [inline src/upload.ts:8]: why do we need to export `IMAGE_PROVIDERS` if we use them only inside `set`?
btw, could be `set` renamed to something like `setProviders`?

File: src/upload.ts [hunk 2]
```diff
@@ -38,12 +35,7 @@ router.post('/upload', async (req, res) => {
         .pipe(transformer)
         .toBuffer();
 
-      const result = await Promise.any([
-        setFleek(buffer),
-        setInfura(buffer),
-        setPinata(buffer),
-        set4everland(buffer)
-      ]);
+      const result = await Promise.any(set(IMAGE_PROVIDERS, buffer));
```
Comment (Todmy) [inline src/upload.ts:38]: can we move `Promise.any` inside the `set` function?
```suggestion
      const result = await setProviders(buffer);
```

File: src/upload.ts (hunk 1 above)
Comment (wa0x6e) [reply src/upload.ts:8]: `set()` takes a list of providers as first arg. It's used also in rpc.ts, where it uses different list of providers

Comment (wa0x6e) [reply src/upload.ts:8]: Renamed to `uploadToProviders`

**Review by Todmy — APPROVED** [review]
utACK

**Pull request merged**